packages/sdk/src/evm/core/wallet/user-wallet.ts

```diff
@@ -112,13 +119,19 @@ export class UserWallet {
         receipt: await tx.wait(),
       };
     } else {
-      // ERC20 token transfer
-      return {
-        receipt: await this.createErc20(resolvedCurrency).sendTransaction(
-          "transfer",
-          [resolvedTo, amountInWei],
-        ),
-      };
+      const transferTransaction = await Transaction.fromContractInfo({
+        contractAddress: resolvedCurrency,
+        contractAbi: ERC20Abi,
+        provider: this.connection.getProvider(),
+        signer,
+        method: "transfer",
+        args: [resolvedTo, amountInWei],
+        storage: new ThirdwebStorage({
```

Comment:
ugh a bit annoying to make it mandatory here, @MananTank react-core level fix didnt work i assume?

since we're doing this might as well do the old flow with ContractWrapper tbh

Reply:
reverted it